### ISSUE TITLE
Provide a construction API returns a tuple

### ISSUE THREAD

**Issue by fishtreesugar** [acalejos/flint#13]
Currently, there are two APIs for constructing a struct: `new` and `new!`. The `new` function ignores changeset errors and returns the struct regardless, while `new!` raises an `ArgumentError` when there are changeset errors.
In Elixir, using `with` for error handling is common:
```Elixir
with {:ok, payload} <- Foo.parse(raw_input),
     :ok, _result} <- do_sth(payload) do
  :ok
end
```
To align with the ["Parse, don't validate"](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/) slogan, it would be beneficial if `flint` provided such an API.

**Comment by acalejos** [comment]
I definitely contemplated this. It should probably change to `new` outputting a tuple, and `new!` raising. Then maybe have an option in `new` like `:ignore_errors`. Thoughts?

**Comment by fishtreesugar** [comment]
> It should probably change to new outputting a tuple, and new! raising. 

Yes, I think it should be closer to the convection of Elixir's stdlib, e.g. `fetch/2` v.s. `fetch!/2`

> Then maybe have an option in new like :ignore_errors. Thoughts?

In my opinion, it shouldn't allow an easy way to skip the error. 😂


FWIW, thank you for your quick response!
